packages/samples/headless-react/src/components/folded-result-list/folded-result-list.class.tsx

```diff
@@ -31,7 +32,13 @@ export class FoldedResultList extends Component<{}, FoldedResultListState> {
     this.setState(this.controller.state);
   }
 
-  private renderFoldedResult(result: FoldedResult) {
+  private isCollection(
+    result: Collection | FoldedResult
+  ): result is Collection {
+    return 'moreResultsAvailable' in result;
+  }
```

Comment:
Seeing such a function in the code sample is a red flag. Someone implementing a folded result should not need to perform such a check. How can we adjust the controller api to make things simpler?

Reply:
Well, perhaps it's the sample that should be adjusted then. I added this check only so that I could recursively use the same function for both collections and folded results, but they are technically different since the collection is always at the top-level, so I could separate them in the sample and remove this check.

In the controller API, I don't really see a simpler implementation. Using the same type for folded results and collections would get extremely confusing since it would imply that `loadCollection`/`loadAll` can be executed on any descendant.

Reply:
> I added this check only so that I could recursively use the same function for both collections and folded results

Indeed, this is a good goal, as a developer, I would prefer to work with a tree that is easier to recur over than harder.

> they are technically different since the collection is always at the top-level

Why do they need to be different? Why not provide the same keys and set their values accordingly?

Reply:
Because not the same operations can be executed on leaves and on the root, and adding those fields to the leaves would either contain wrong information or redundant information. Having the same keys on both would give a very misleading impression. For the next PR, I'll try to draw a diagram.